impress-org/givewp
# Multi-form Goal block is ready for alpha

## Description
This PR represents work done so far on the Multi-Form Goals epic. At this point, the work here is ready for an alpha release.

Specifically, this Epic introduces the Multi-Form Goal block and shortcode.

## Affects
This PR introduces the MultiFormGoals domain, where all logic related to editing and displaying the Multi-Form Goals block and shortcode can be found.

## Visuals

Screencast walkthrough:

https://www.loom.com/share/f88c1de9d7f04b49b8ac9611594f8a57

## Pre-review Checklist

## Testing Instructions
1. Add a new page or post
2. Add a Multi-Form Goal block
3. Modify various settings, and see that changes are reflected in the editor and on the front end
4. Add a shortcode block, first with a bare `[give_multi_form_goal]`shortcode.
5. Check that the shortcode renders as expected, then try modifying different shortcode attributes (`heading`, `summary`, `image`, `color`, `goal`, `enddate`, `tags`, `categories`, `ids`)

Note: The default copy used for the shortcode needs to be improved. For an example of a shortcode with most attributes filled out, see:
`[give_multi_form_goal heading="Test Heading" summary="Test summary!" goal="250" color="#ddd" enddate="2020-12-10" ]`

## User Documentation
This Epic needs user documentaiton to support the new Mutli-Form Goal block and its corresponding shortcode.
